A New Governance Structure: SEP 2 Revision - Final Comment Period (#80)
sunpy/sunpy-SEP: SEP-0002.md

Whenever possible, decisions shall be approved by consensus, considering input from the Community, rather than taken unilaterally by Project Custodians or the Steering Committee.

### The Project Custodians

The Project Custodians are trusted community members with a proven track record of constructive contribution to the goals of the project.
Collectively, the Project Custodians make decisions about the direction of the project and have authority over the project, which is delegated to the Steering Committee.

It is expected that Project Custodians act as role models for the Community and are custodians of the SunPy Project.
The Project Custodians elect a Steering Committee to lead the SunPy Project.

#### Responsibilities and Rights

Project Custodians are granted voting rights on matters related to how the project is run and organized and can elect (and remove) project officers. Some examples are listed below.

* Adding new Project Custodians
* Electing Steering Committee members
* Appointing the Ombudsperson
* Removing Project Custodians
* Removing Steering Committee members

> * Removing the Ombudsperson

Author: ```suggestion
* Removing the Ombudsperson
* Actively solicit feedback from and engage with the community to make decisions they believe are in the best interests of The Community
```

| SEP           | 2                                                       |
|---------------|---------------------------------------------------------|

> | title         | SunPy Organization Definition                           |

Author: I am not sure how well your suggested title would age, like in 5 years is this really the "updated" definition?

| author(s)     | [Stuart Mumford](https://orcid.org/0000-0003-4217-4642) |
| contact email | <stuart@cadair.com>                                     |
| date-creation | 2024-04-09 (Original version: 2014-02-16)               |
| type          | process                                                 |

> | discussion    | <https://github.com/sunpy/sunpy-SEP/pull/62>            |

Reviewer: Add #80 as well?

| status        | in review                                               |

## Rationale

This SEP describes the second major iteration of the governance for the SunPy Project.

> The objectives of the rewrite were to adapt the governance for the changes the project has seen since the first iteration of this document was written in 2014.

Author: How's this?
```suggestion
The objectives of this update are to optimize the governance based on our experiences running the project since 2014 when the original governing SEP was written. That original governance structure was modeled after a non-profit company structure, assumed a very committed board, and intertwined project finances with board responsibilities.  The project has since matured and responsibilities for project management should be fully given over to the community while a smaller board with a reduced mandate provides high-level oversight.
```

The rewrite aimed to detail a governance structure, which strives to be lightweight, while being designed to enable the project to grow, and facilitating more input from the wider SunPy community.

## The SunPy Project

> The primary mission of the SunPy Project is **To develop and facilitate a high-quality, easy-to-use, community-led, free and open-source solar data analysis ecosystem based on the scientific Python environment**.

Reviewer 2: In my reading, this mission statement aims to support (in a general sense, not necessarily in coding or technical maintenance) all uses of Python in solar data analysis, not only those uses based on SunPy. 
But I know it was an effort to wordsmith this mission statement in the first place, so I'm also fine leaving it be.

This includes but is not limited to the following tasks:

* developing the sunpy core library and other affiliated packages
* supporting a community of affiliated packages
* managing and protect the SunPy brand and identity
* promoting SunPy to the community

> * managing SunPy assets, i.e. GitHub organization, domain name etc.

Reviewer: ```suggestion
* managing SunPy assets, e.g., GitHub organization, domain name etc.
```

## Governance Structure

The SunPy Project's governance involves the following five components:

* The Community

> * The Project Custodians

Author: > I am not a fan of the term Project Custodian.

We litigated this to death on the old PR so I am not willing to reopen that discussion: https://github.com/sunpy/sunpy-SEP/pull/62#discussion_r1122282963

> I suggest that the Steering Committee description be removed into a different SEP in order to make it easier for Project Custodian to change how the leadership is organized.

I am not a fan of this solely due to how much of a pain it has been to get this SEP to this point. While I see your point, I don't think it's a deal breaker and also I feel like changing this should require a high-bar of custodian votes.

* The Steering Committee
* The Advisory Board
* The Ombudsperson

The SunPy project exists for the benefit of the whole Community: users, developers, and members of the wider solar physics and scientific software community.
The Community is the ultimate authority in how the project is managed.
This authority is exercised through the processes outlined in this document, primarily through the Steering Committee and the Project Custodians.

> Whenever possible, decisions shall be approved by consensus, considering input from the Community, rather than taken unilaterally by Project Custodians or the Steering Committee.

Reviewer 2: first cut:
"When possible and necessary, input will be gathered from the Community through multiple input channels, including online polling, direct communication, and in person interactions at online gatherings. These input channels will be actively used to reach decisions by consensus, rather than taken unilaterally by Project Custodians or the Steering Committee."

> Whenever possible, decisions shall be approved by consensus, considering input from the Community, rather than taken unilaterally by Project Custodians or the Steering Committee.

Author: ```suggestion
When feasible and necessary, input will be gathered from the Community through multiple channels, for example forums, online polling, direct communication, and in-person interactions at domain-related meetings.
Wherever possible, these input channels will be actively used to reach decisions by consensus, rather than taken unilaterally by Project Custodians or the Steering Committee.
```

The Project Custodians elect a Steering Committee to lead the SunPy Project.

#### Responsibilities and Rights

Project Custodians may and should participate in formal votes on:

* Adding new Project Custodians
* Electing Steering Committee members
* Appointing the Ombudsperson
* Removing Project Custodians
* Removing Steering Committee members
* Removing the Ombudsperson
* Approving changes to this SEP via a vote (see below)
* Other matters where the Steering Committee or the Advisory Board believes a vote is appropriate.

#### Membership

Project Custodians of the SunPy Project must demonstrate all of the following:

* A good grasp of the philosophy of the SunPy Project
* A solid track record of being constructive and helpful
* Significant contributions to the Project's goals, in any form
* A willingness to dedicate time to improving the Project
* A willingness to recruit, train, and nominate new team members
* A track record of adhering to the project Code of Conduct.

Project Custodianship acknowledges sustained and valuable efforts that align well with the philosophy and the goals of the SunPy Project.

#### Initial Membership

> To be able to elect new Project Custodians an initial set of Project Custodians are needed.

Reviewer: ```suggestion
To be able to elect new Project Custodians an initial set of Project Custodians is needed.
```

People meeting the following criteria will be invited to form the initial group of Project Custodians.
They should consider the above responsibilities and actively agree to taking them on if they choose to accept the invitation.

* Current commit access to the core repo, or any accepted affiliated package maintained by the project, and with a commit to that repository within the last 12 months.

> * Currently holds a role listed under the community roles section of the website: <https://sunpy.org/project/##community-roles>.

Reviewer: ```suggestion
* Currently holds a role listed under the community roles section of the website: <https://sunpy.org/about/project/#community-roles>.
```

* Is a currently serving Advisory Board member.

Applying this set of criteria on 2024/04/09 resulted in the following people being the initial Project Custodians:

* Stuart Mumford
* Will Barnes
* Laura Hayes
* David Pérez-Suárez
* Conor MacBride
* Monica Bobra
* Steven Christe
* Russell Hewett
* David Perez-Suarez
* Tiago M. D. Pereira
* Kevin Reardon
* Albert Y. Shih
* Nabil Freij
* David Stansby
* Alasdair Wilson
* Daniel Ryan
* Shane Maloney

#### Adding New Project Custodians

Anyone is eligible to be nominated to be a Project Custodian, as long as they meet the criteria laid out in the Membership section above.
Self-nominations are encouraged.

> The process for adding a new Project member is as follows:

Reviewer: ```suggestion
The process for adding a new Project Custodian is as follows:
```

> 1. A nomination is received by the Steering Committee, a nomination should include a short, factual, statement stating how the nominee meets the criteria for Project Custodianship.

Reviewer 3: I think when rendered they are incremented?

> 1. A nomination is received by the Steering Committee, a nomination should include a short, factual, statement stating how the nominee meets the criteria for Project Custodianship.

Reviewer 4: Ah ok. I'm just looking at the raw markdown (on a phone, just how it was meant to be read).

The Steering Committee, and the Ombudsperson check that the nomination is factually accurate, their record of contribution to the community adheres to the Code of Conduct, and they wish to serve as a Project Custodian (if not self-nominated).
1. The candidate is then put to the Project Custodians.
The name and nomination statement of the candidate will be shared with the Project Custodians.

> The candidate can become a Project Custodian if, during the two weeks following the candidate being put to the Project Custodians, five existing Project Custodians approve the candidate and no Project Custodians call for an election.

Author: It does not.

> The candidate can become a Project Custodian if, during the two weeks following the candidate being put to the Project Custodians, five existing Project Custodians approve the candidate and no Project Custodians call for an election.

Author: We give up and go home?

> The candidate can become a Project Custodian if, during the two weeks following the candidate being put to the Project Custodians, five existing Project Custodians approve the candidate and no Project Custodians call for an election.

Reviewer 5: Sounds good.

1. Project Custodians can call for an election either by informing the Steering Committee or the Ombudsperson.
If a vote is called for then the candidate will become a Project Custodian with a simple majority of all votes cast in the election.
An election needs to be announced to the Project Custodians one week before the voting opens and voting will be open for two weeks.
1. The candidate will be informed promptly of the outcome by the Steering Committee.

The name of a candidate should not be shared outside of the Project Custodians until they become a Project Custodian.
A public list of Project Custodians will be kept on the SunPy website.

Nominations, approvals and any elections of Project Custodians should take place before the initial election of the Steering Committee.

#### Active and Emeritus Status

Project Custodians have no term limits.

> Project Custodians who are no longer actively contributing to the project should contact the Steering Committee to move themselves to emeritus status.

Reviewer: This section is a bit frustrating to read because "active" as a status is not clearly defined and delineated, so you end up implicitly discussing non-active (in terms of contributions) active (in terms of status) custodians and active (in terms of contributions) non-active (i.e., emeritus) custodians.

Project Custodians who have not participated in the project in the manner required for Project Custodianship for two years, will be contacted by the Steering Committee and be asked to move themselves to emeritus status.
The Steering Committee can change a Project Custodian's status to emeritus.
Emeritus Project Custodians will continue to be listed on the website.
Emeritus Project Custodians are not able to participate in votes.
An emeritus member may declare themselves to be active again at any time if they feel they are making contributions.

#### Removing Project Custodians

In exceptional circumstances, it may be necessary to remove someone from the Project Custodians against their will.

> A vote of the Project Custodians must be held to remove a Project Custodian.

Author: How would you suggest that could work? The whole point of the custodian role is that they are "trusted" community members who make decisions on behalf of the wider community.

> Such a vote is triggered by a motion (sent to the Steering Committee and the Ombudsperson) made by an active Project Custodian, which must be seconded by an additional Project Custodian.

Reviewer: Per my comment above, it can be unclear whether "active" here is referring to status (i.e., non-emeritus) or merely to an engaged custodian.

Removal requires approval by two-thirds of all (non-emeritus) Project Custodians at the time the motion is made.
The motion, second, and vote will be by secret ballot.
Removal under this provision will be reflected by updating the list of Project Custodians.
A member removed via this mechanism does not have emeritus status.

It may be necessary for the Ombudsperson and the Steering Committee to remove a Project Custodian for violations of the Code of Conduct.
In this case, the Steering Committee, Ombudsperson and Advisory Board will work together to make this decision.

#### Votes on Other Matters

From time to time, as desired or required by the Steering Committee and Advisory Board, matters may be put to the Project Custodians for a vote.
For these votes, the motion will carry by a simple majority of those Project Custodians who cast a ballot.

### The Steering Committee

The role of the Steering Committee, which is chaired by the Lead Developer, is to lead the day-to-day operations of the project.
The Steering Committee shall be responsible for the running of the project under advisement of the Advisory Board.
This includes:

> * holding regular meetings with the community (at least monthly)

Author: ```suggestion
* holding regular meetings with the community (at least monthly) 
* publicizing project efforts and status through community-appropriate channels and social media
```

* making regular reports to the Advisory Board (at least quarterly)
* having ownership privileges to the SunPy GitHub organization and other community assets
* creating, maintaining and enforcing the SunPy code of conduct (see below)
* accepting or rejecting SEPs based on community consensus (excluding this SEP, see below)
* delegating responsibility and authority to other community roles
* planning and managing finances and financial processes of the project.
* Ensuring SunPy adheres to its legal obligations, such as to its fiscal sponsor.

#### Composition

> The Steering Committee is formed of three members of the SunPy community with equal status on the committee, and they do not need to be Project Custodians.

Reviewer: If you want an analogy in U.S. politics, one would be how the House of Representatives can - in theory - elect a non-member to be Speaker of the House.  Of course, this has never happened to date.

> The Steering Committee is formed of three members of the SunPy community with equal status on the committee, and they do not need to be Project Custodians.

Reviewer: Anyway, given the managerial roles that are being delegated to the Steering Committee, I think it is plausible for a good Steering Committee member to not actually be a good Project Custodian, namely on these two criteria for the latter:

> * A solid track record of being constructive and helpful
> * A willingness to recruit, train, and nominate new team members

I don't see why a Steering Committee member ought to be required to be either of those, although they are of course positives.  If the collective wisdom of the Project Custodians is to elect a non-Custodian, I say so be it.

> The Steering Committee is formed of three members of the SunPy community with equal status on the committee, and they do not need to be Project Custodians.

Author: I also don't see why we *need* to require it, it can be left to the custodians on a case-by-case basis?

The Steering Committee can decide how to distribute the work load, however the Steering Committee should self-appoint the following roles after each election:

* **The lead developer** chairs the Steering Committee and is responsible for holding regular (at least monthly) meetings, and ensuring that the committee is represented at all Advisory Board meetings.
* **The financial officer** the financial officer is primarily responsible for ensuring the health of the project finances and setting up processes for managing expenses and reporting to the Advisory Board on financial matters.

The Steering Committee is elected by the Project Custodians.

#### Decision Making

The Steering Committee should strive to make decisions by consensus.
Where possible and appropriate, consensus should be reached between the members of the committee, and that consensus should reflect the consensus of the whole Community.

> If consensus can not be reached between the committee members, decisions can be taken by a majority vote.

Author: It means they are responsible for organising the meetings and following the agenda. That's it. Really it's an excuse to keep the "lead developer" title in operation.

For the member(s) to be removed, two-thirds of the (non-emeritus) Project Custodians (where the count includes the member being voted on if they are a Project Custodian) must vote in support of the removal.

To avoid repeated removal votes, any individual Project Custodian who has called for, or seconded such a vote, may not call for or second a vote to remove any member(s) for one year from the original motion.

### Advisory Board

The primary functions of the Advisory Board are:

* to advise the Steering Committee,
* aid in finding funding for the project to ensure the core aims of the project can be met,
* nominate a Ombudsperson,
* ensure that the Project is being run consistent with the approved governance structure and the needs of the Community.

The Advisory Board is self-electing, with members appointed by a vote of the existing Advisory Board.

#### Membership

* The Advisory Board shall consist of at least 3 members and no more than 5 members. The Advisory Board selects how many members within this range are needed.
* All members shall serve a maximum of a two-year term, and may be re-appointed at the end of their term.

> * Membership terms should be staggered so that approximately half the number of members will end their terms in any given year.

Author: ```suggestion
```

* The Advisory Board may appoint members to its number up to the maximum number of members.

> * The Advisory Board should work to ensure that the composition of the Advisory Board represents a diverse cross-section of the scientific community with an interest in SunPy. This diversity should consider things such as research topic, career stage, as well as all other forms of diversity.

Author: I think this is more relevant to the board as they self-elect members. The steering committee is elected by the custodians, so it would be hard to place a requirement like this on that. We could add something to the custodians in terms of  "encouraging nominations of people from a diverse section of the community", but really *anyone* who meets the requirements should be a custodian. The challenge in diversity there is encouraging the participation in the community in the first place?

* An Advisory Board member may be removed by a two-thirds majority vote of the Advisory Board or by Project Custodians following the same procedure as for the removal of a member of the Steering Committee.

#### Meetings

* The Advisory Board shall meet with the Steering Committee at least every quarter.
* At least one week (7 days) notice of a meeting shall be provided to all eligible to attend.
* Notice of meetings shall provide the information necessary to participate in the meeting.
* Minutes for the meetings shall be recorded and made public as much as is appropriate.

The Advisory Board shall organize itself as it sees fit.

#### Advise and Consent

The Steering Committee shall present the state of the project and future plans at meetings of the Advisory Board on a yearly basis.
The Advisory Board shall advise and consent on those plans.

#### Requiring a Project Vote

In the unlikely event that the Advisory Board does not agree with a decision made by the Steering Committee, it can require Project Custodians to consent to the decision through a vote (see below for process).

> ## Ombudsperson

Author: ```suggestion
### Ombudsperson
```
